src/VCAuthn/IdentityServer/QueryStringSecretParser.cs

```diff
@@ -0,0 +1,81 @@
+using System.Linq;
+using System.Threading.Tasks;
+using IdentityModel;
+using IdentityServer4;
+using IdentityServer4.Configuration;
+using IdentityServer4.Models;
+using IdentityServer4.Validation;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+
+namespace VCAuthn.IdentityServer
+{
+    public class QueryStringSecretParser : ISecretParser
+    {
+        private readonly IdentityServerOptions _options;
+        private readonly ILogger<QueryStringSecretParser> _logger;
+
+        public QueryStringSecretParser(IdentityServerOptions options, ILogger<QueryStringSecretParser> logger)
+        {
+            _options = options;
+            _logger = logger;
+        }
+        
+        public string AuthenticationMethod => "client_secret_query";
+
+        public async Task<ParsedSecret> ParseAsync(HttpContext context)
+        {
+            _logger.LogDebug("Start parsing for secret in request query");
+
+            var query = context.Request.Query;
+
+            if (query == null)
+            {
+                _logger.LogDebug("No secret in query found");
+                return null;
+            }
+
+            var id = query["client_id"].FirstOrDefault();
+            var secret = query["client_secret"].FirstOrDefault();
+
+            // client id must be present
+            if (string.IsNullOrEmpty(id))
+            {
+                _logger.LogDebug("No client in query found");
```

Comment:
No client_id in query found

Reply:
Messages are consistent with IdentityServer implementations of ISecretParser

Reply:
Actually, yeah, will change both this and the "maximum length" message.